src/app/cake/cake.class.ts

```diff
@@ -0,0 +1,46 @@
+import { iCake } from './cake.interface';
+import { CakeStatus } from './cake-status.enum';
+import { dbConn } from './../app.database';
+import { CakeService } from './cake.service';
+
+export class Cake implements iCake {
+  private cakeService: CakeService;
+
+  name: string;
+  description: string;
+  ingredients: string[];
+  price: number;
+  stock: number;
+  status: CakeStatus;
+
+  constructor(cake: iCake) {
```

Comment:
It would be a better idea to inject your services, if you do it this way you would be able to inject a mock of these services for test purposes.

```
type CakeInjection = { cakeService: CakeService };
...
constructor(cake: iCake, injection: CakeInjection) {
  this.cakeService = injection.cakeService || new CakeService();
  ...
}
```

Reply:
I see! Thanks, I will change this